packages/frontend/src/pages/Transaction/TransactionPage.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { useEffect, useState } from 'react'
+import { useParams } from 'react-router'
+import { Div, Flex } from 'src/components/ui'
+import { useTransaction } from 'src/hooks'
+import SmallTextField from 'src/components/SmallTextField'
+import { Link } from '@material-ui/core'
+import { useApp } from 'src/contexts/AppContext'
+import Network from 'src/models/Network'
+import { networkIdToSlug, truncateHash } from 'src/utils'
+import { utils } from 'ethers'
+import { Text } from 'src/components/ui/Text'
+import { getExplorerTxUrl } from 'src/utils/getExplorerUrl'
+import { metadata } from 'src/config'
+import { useWeb3Context } from 'src/contexts/Web3Context'
+import { NetworkSelector } from 'src/components/NetworkSelector'
+import { DisplayObjectValues } from './DisplayObjectValues'
+
+function TransactionPage() {
+  const { hash } = useParams<{ hash: string }>()
+  const [txHash, setTxHash] = useState<string>(hash)
+  const [sourceNetwork, setSourceNetwork] = useState<Network>()
+
+  const { sdk, tokens } = useApp()
+  const { connectedNetworkId } = useWeb3Context()
+
+  useEffect(() => {
+    if (hash) {
+      setTxHash(hash)
+    }
+  }, [hash])
+
+  // STATE
+  // balances
+  // liquidity
+  // tx types: approve / sendL1ToL2 / sendL2ToL2 / sendL2ToL1 / convert / wrap / unwrap / addLiquidity / removeLiquidity / stake / unstake?
+  // link to graph
+  // add token to MM
+
+  const { txObj, loading, error, confirmations, networkConfirmations } = useTransaction(
+    txHash,
+    sourceNetwork?.slug
+  )
+
+  useEffect(() => {
+    console.log(`txObj:`, txObj)
+  }, [txObj])
+
+  async function handleSetNetwork(network) {
+    if (txHash && network) {
+      setSourceNetwork(network)
+    }
+  }
+
+  async function addToken() {
+    if (txObj?.tokenSymbol) {
+      console.log(`tokens['DAI']:`, (tokens as any).DAI)
+      const { tokenSymbol } = txObj
+      const tokenImageUrl = metadata.tokens[tokenSymbol].image
+      const tokenModel = sdk.toTokenModel(txObj.tokenSymbol)
+      const networkName = networkIdToSlug(connectedNetworkId)
+      if (networkName === 'ethereum') {
+        return
+      }
+      console.log(`tokenImageUrl:`, tokenImageUrl)
+      const addr = tokens[tokenSymbol][networkName].l2CanonicalToken.address
```

Comment:
since the sdk is already being imported here, then we can call the sdk method to get canonical token address here instead of doing map lookup

```typescript
const addr = sdk.getL2CanonicalTokenAddress(tokenSymbol, networkName)
```